packages/maskbook/src/plugins/ITO/UI/ITO.tsx

```diff
@@ -308,6 +308,18 @@ export function ITO(props: ITO_Props) {
         },
     )
 
+    useEffect(() => {
+        const timeToExpired = end_time * 1000 - new Date().getTime()
+        if (timeToExpired < 0 || listOfStatus.includes(ITO_Status.expired)) return
+
+        const timer = setTimeout(() => {
+            setOpenClaimDialog(false)
+            retryITOCard()
+        }, timeToExpired + TIME_WAIT_BLOCKCHAIN)
```

Comment:
Does this duration represent the average blocking time of the Ethereum chain?

Reply:
No, it is a conservative time to ensure expired.